docs/hackathons/ihp_10_2022.rst

```diff
@@ -0,0 +1,285 @@
+
+
+Hackathon at Institut Henri Poincaré, 17th-21st Oct 2022
+========================================================
+
+This hackathon happened at IHP within the thematic trimester on `Geometry and Statistics in Data Sciences <https://indico.math.cnrs.fr/category/498/>`_.
+
+
+Several participants made their first contributions to ``geomstats``. In no particular order: Francesco Ballerin (University of Bergen), Juliane Braunsmann (WWU Munster), Antoine Collas (Paris-Saclay University), John Harvey (Cardiff University), Emmanuel Hartman (Florida State University), and Joan Glaunès (Université Paris Descartes).
+
+
+Besides the usual informal schedule, we had a very fruitful presentation by Jean Feydy (INRIA), entitled "Fast geometric libraries for vision and data sciences". 
+
+
+It was a prolific week in terms of (great) discussions about math and programming. Several actions to improve our code base originated from them. The next section describes them and the rationale behind the changes that will be implemented.
+
+
+
+Future changes
+--------------
+
+
+Equip with metric
+*****************
+
+The following way of instantiating a space and equipping it with a metric will become standard:
+
+
+.. code-block:: python
+
+    from geomstats.geometry.hypersphere import (
+        Hypersphere,
+        HypersphereMetric,
+    )
+
+    space = Hypersphere(dim=2, equip=False)
+    space.equip_with_metric(HypersphereMetric)
+
+
+This means metrics are no more objects that should be instantiated by themselves, i.e. metrics are associated with spaces. To use methods available within the metric, it is now preferred to do ``space.metric.<method>`` instead of simply instantiating a metric and do ``metric.<method>``.
+
+
+Mathematically, this means that ``space.equip_with_metric`` transforms our smooth manifold :math:`M` into the tuple :math:`(M, g)`. The boolean ``equip`` allows to have the possibility to instantiate a non-equipped manifold. In order to keep the current behavior and for practical reasons (e.g. it can be cumbersome to equip a space that only has one "valid" metric; it is nice to have embedding spaces already equipped), ``equip=True`` by default, which means the following code is equivalent to the above:
+
+
+.. code-block:: python
+
+    space = Hypersphere(dim=2)
```

Comment:
If I understand correctly, the metric is instantiated in `space.equip_with_metric(HypersphereMetric)`. Is this called by `__innit__` when `equip=True`, but this does not create circularity issues because `space=self` in the metric?

Reply:
Exactly!